lib/DateTime/Format/Strptime.pm

```diff
@@ -986,6 +986,9 @@ sub format_datetime {
     my $self = shift;
     my $dt   = shift;
 
+    croak 'format_datetime() expects parameter an object of type DateTime.'
```

Comment:
I think it'd make more sense to just use `Params::ValidationCompiler` to do the validation and let it generate the appropriate error message.